Issue #3181482 by rolki: As a LU I want to have clear overview on each activity item on dashboard stream

rollki opened the pull request:
## Problem
The activity stream items are not readable. It means that body text in activity stream posts and comments can be too long (not trimmed), also there's comment with a commented entity, ideally, it should be comment without commented entity and comment from there also 😑

See discussion here: https://getopensocial.atlassian.net/browse/YANG-4059

## Solution
Improve UI/UX, remove unneeded items (commenting form, commented entity, and etc). 
See design proposal https://getopensocial.atlassian.net/browse/YANG-4059 

At the moment design proposal was changed a bit since we have merged the album feature and a lot of other things now.

## Issue tracker
- https://www.drupal.org/project/social/issues/3181482
- https://www.drupal.org/project/socialbase/issues/3198471
- https://getopensocial.atlassian.net/browse/YANG-5826

## How to test
- [ ] Use this PR as a patch
- [ ] Use `social_dashboard` extension with https://bitbucket.org/goalgorilla/social_dashboard/pull-requests/71 changes
- [ ] Apply https://git.drupalcode.org/project/socialbase/-/merge_requests/2 changes for `socialbase`
- [ ] Create a new dashboard with an activity stream
- [ ] You should see beautiful activity items with a new design

## Screenshots
### Before
![image](https://user-images.githubusercontent.com/50984627/104465957-50efba00-55bd-11eb-8825-f009c4a82638.png)
![image](https://user-images.githubusercontent.com/50984627/104466581-0f134380-55be-11eb-8229-4ffe72d195a0.png)

### After
![image](https://user-images.githubusercontent.com/50984627/124087081-57ce9400-da5a-11eb-9dbf-83a7a0971169.png)
![image](https://user-images.githubusercontent.com/50984627/124087124-64eb8300-da5a-11eb-982d-844ae2f1cf5a.png)


## Release notes
This change doesn't affect distro work, we just move some code that related to the dashboard to an extension instead of here.

ribel commented on modules/custom/activity_logger/activity_logger.tokens.inc:
```diff
@@ -146,12 +151,18 @@ function activity_logger_tokens($type, $tokens, array $data, array $options, Bub
                     $group = $group->getTranslation($curr_langcode);
                   }
-                  $replacements[$original] = $group->label();
+
```
@rollki Please add explanation comment on what we do here.

rollki replied:
Added description.

ribel commented on modules/social_features/social_comment_upload/social_comment_upload.module:
```diff
@@ -71,13 +72,17 @@ function social_comment_upload_preprocess_comment(&$variables) {
         // Add the amount of attachments to the comment.
         if (!empty($count_arr)) {
-          $text = \Drupal::translation()->formatPlural(count($count_arr),
-            '1 attachment.',
```
@rollki If we remove some strings we need to add them to `translations.php` to keep translations for older versions.

ribel commented on modules/social_features/social_comment_upload/social_comment_upload.module:
```diff
@@ -71,13 +72,17 @@ function social_comment_upload_preprocess_comment(&$variables) {
-            '@count attachments.'
-          );
-          $variables['comment_attachments_count'] = Link::fromTextAndUrl($text, $comment->permalink());
+          $variables['comment_attachments_count'] = count($count_arr);
```
@rollki could you explain what changes we do here and why?

ribel commented on modules/social_features/social_comment_upload/social_comment_upload.module:
```diff
@@ -71,13 +72,17 @@ function social_comment_upload_preprocess_comment(&$variables) {
+        $variables['comment_url'] = $topic->toUrl()->toString();
+      }
+
+      $more_button = $comment->toLink('Read more')->toString();
```
@rollki Is this new string `Read more` translatable?

rollki replied:
Added `t` function

ribel commented on themes/socialbase/templates/comment/comment--activity.html.twig:
```diff
@@ -66,9 +66,8 @@
-{{ content.commented_entity }}
-{{ content.comment_link }}
-<div class="card__nested-section">
+
```
@rollki Did we check if after removing those this kind of comments looks good in default activity streams in both themes?

ribel commented on themes/socialbase/templates/comment/comment--activity-comment.html.twig:
```diff
@@ -1,7 +1,60 @@
-     <span class="comment__metadata">&bullet; {{ submitted }} </span>
-   </div>
- {% endblock %}
+{%
```
@rollki Did we check if after removing those this kind of comments looks good in default activity streams in both themes?

ribel reviewed (changes requested):
@rollki I know this feature looks already good in Dashboard, but before merging it o a Distro, I have few questions, which would be good to clarify - please check.

ribel commented:
@rollki Also please fix coding standards and add some more screenshots which show before/after visual changes.

rollki reviewed (commented):
@ribel Checked via drupal code sniffer - everything fine. Screenshots added and answered to all comments.

ribel commented:
@rollki Behat test `activity-stream-comments` is failing - looks like this is related to changes in this PR, so please check and fix.

ribel commented:
@rollki I see there are failing tests related to `SocialPostPhotoConfigOverride` - please check.

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

rollki commented:
Things that should be discussed:
* Trimmed text module
* Message template (gtrimmed text)

rollki commented:
Hey @ronaldtebrake 
We allotted time to finish this PR, but there are some things that need to be agreed upon.
So we install here new module `smart_trim`, and we used this module to trim description in featured view mode, it is also possible to trim via default trim functionality, but this module provides the ability to add a suffix, in our case it is `...`, and it looks like:
![image](https://user-images.githubusercontent.com/50984627/121521952-4ff07680-c9fd-11eb-96ef-48e0b3a12edc.png)
**So, we need consent here :)**

Another thing is message templates, we added a new token here's `gtrimmed-title`, we added this token, because this is the easiest way to trim group titles, and not sure if it is possible by another way.
![image](https://user-images.githubusercontent.com/50984627/121522450-ca20fb00-c9fd-11eb-9e51-835fca5f683a.png)
So, we also need consent here, if you think this is a bad idea to change message templates, then we can scrap it but note that we have clients that have too long group names, and it was actually requested from clients to trim group title.

cc @ribel

rollki commented:
BTW, on hover it is possible to see full group name.
![Снимок экрана 2021-06-10 в 15 12 03](https://user-images.githubusercontent.com/50984627/121522863-3bf94480-c9fe-11eb-9fd4-6b89046744ec.png)

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

Kingdutch commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

ribel commented on modules/custom/activity_logger/activity_logger.tokens.inc:
```diff
@@ -28,6 +28,10 @@ function activity_logger_token_info() {
     'description' => t("The related node title."),
   ];
 
+  $message['gtrimmed-title'] = [
```
@rollki I would recommend not introduce a new token for group titles.
Maybe you can check and use `hook_tokens_alter()` in the dashboard module and trim the existing `gtitle` when the Dashboard module is enabled?

rollki replied:
As we discussed we don't want it yet, so I have removed changes related to the trimmed group title, in the future when it will be necessary we will be able to add this feature.

ribel commented on composer.json:
```diff
@@ -163,6 +163,7 @@
         "drupal/socialblue": "^1.0",
         "drupal/swiftmailer" : "2.0-beta1",
         "drupal/select2": "1.8",
+        "drupal/smart_trim": "1.3",
```
@rollki I think it's better to add this depdendcy in the Dashboard module.

rollki replied:
Sure :)

rollki commented on modules/social_features/social_post/social_post.module:
```diff
@@ -207,26 +207,6 @@ function social_post_theme() {
-/**
- * Implements hook_preprocess_HOOK().
- */
-function social_post_preprocess_post(&$variables) {
```
Preprocess was moved to the `social_challange` module.

rollki commented on modules/social_features/social_comment/social_comment.module:
```diff
@@ -291,26 +291,10 @@ function social_comment_comment_view(array &$build, EntityInterface $entity, Ent
     if (is_object($commented_entity)) {
       $commented_entity_type = $commented_entity->getEntityTypeId();
 
-      // Define which display mode should be used for commented_entity.
```
The logic was moved to the `social_challange` module, so we don't need these conditions here anymore.

rollki commented on modules/social_features/social_post/templates/post--featured.html.twig:
```diff
@@ -53,8 +53,14 @@
           <div class="media-post-social">
             {{ content.like_and_dislike }}
+
             {% if dashboard and comment_count > 0 %}
```
There is no way to override templates in `social_dashboard` :( So, this is the reason why we have conditions to check the `dashboard` variable.

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

rollki commented:
These changes affect the only dashboard, so group, user, landing page streams works as before. 

Not sure if there is any reason to implement similar for user and group stream since there is a bit different view, and there's logical to display a list of comments under the post and etc. But this case can be discussed.

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2060-sygigkdg8cxx6iwzqyecos7wexkagwfo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60c1f7cecde720c4177bb44a

ribel commented:
🍒 picked to 10.2.x
